Compilation error on Ubuntu 20.10

When compiling `//tensorflow_io/core:python/ops/libtensorflow_io.so`, it throws an error:
```
external/avro/impl/Zigzag.cc:66:1: error: 'size_t' does not name a type
   66 | size_t
      | ^~~~~~
```

gcc version
```
gcc (Ubuntu 10.2.0-13ubuntu1) 10.2.0
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Releated to https://github.com/apache/avro/pull/837

I think we need to bump `avro` to at least `1.10.1`.

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -80,11 +80,11 @@ http_archive(
 http_archive(
     name = "avro",
     build_file = "//third_party:avro.BUILD",
-    sha256 = "e382ac6685544ae9539084793ac0a4ffd377ba476ea756439625552e14d212b0",
-    strip_prefix = "avro-release-1.9.1/lang/c++",
+    sha256 = "8fd1f850ce37e60835e6d8335c0027a959aaa316773da8a9660f7d33a66ac142",
+    strip_prefix = "avro-release-1.10.1/lang/c++",
     urls = [
-        "https://storage.googleapis.com/mirror.tensorflow.org/github.com/apache/avro/archive/release-1.9.1.tar.gz",
-        "https://github.com/apache/avro/archive/release-1.9.1.tar.gz",
+        "https://storage.googleapis.com/mirror.tensorflow.org/github.com/apache/avro/archive/release-1.10.1.tar.gz",
+        "https://github.com/apache/avro/archive/release-1.10.1.tar.gz",
     ],
 )
 

